src/Graph/Trees/Heap.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace Rubix\ML\Graph\Trees;
+
+use InvalidArgumentException;
+
+/**
+ * Heap
+ *
+ * A heap is a specialized tree-based data structure that satisfies the heap property:
+ * in a max heap, for any given node C, if P is a parent node of C, then the key of P
+ * is greater than or equal to the key of C. In a min heap, the key of P is less than or equal to the key of C.
+ *
+ * References:
+ * [1] Black (ed.), Paul E. (2004-12-14)
+ * <https://xlinux.nist.gov/dads/HTML/heap.html>
+ *
+ * @category    Machine Learning
+ * @package     RubixML
+ * @author      Rich Davis
+ */
+class Heap
```

Comment:
This is a heap queue, more commonly known as a priority queue, there is no min or max heap. 

If we're referring to how the pop method operates, it will return the smallest item, not the largest, this is what most refer to as a min heap which we can reference.

Reply:
So how does the name HeapQueue sound?